homeassistant/components/media_source/local_source.py

```diff
@@ -56,32 +37,81 @@ def __init__(self, hass: HomeAssistant):
     @callback
     def async_full_path(self, source_dir_id, location) -> Path:
         """Return full path."""
-        return self.hass.config.path("media", location)
+        return Path(self.hass.config.media_dirs[source_dir_id], location)
+
+    @callback
+    def async_parse_identifier(self, item: MediaSourceItem) -> Tuple[str, str]:
+        """Parse identifier."""
+        if not item.identifier:
+            source_dir_id = ""
+            location = ""
+
+        else:
+            source_dir_id, location = item.identifier.lstrip("/").split("/", 1)
```

Comment:
We shouldn't still have to strip `/`. We shouldn't have a `/` there to begin with.

Reply:
The format is `<source_dir_id>/<location>` where `<location>` can include `/`

Reply:
Yes, however, this was needed due to a bug in the URI regex. Will adjust this now.